BiLSTM_CRF.py

```diff
@@ -0,0 +1,126 @@
+import torch
+import torch.nn as nn
+from utils import *
+from torch.nn import init
+
+START_TAG = "<START>"
+STOP_TAG = "<STOP>"
+
+
+def log_sum(smat):
+    max_score = smat.max(dim=0, keepdim=True).values
+    return (smat - max_score).exp().sum(axis=0, keepdim=True).log() + max_score
+
+
+class BiLSTM_CRF(nn.Module):
+    def __init__(self, vocab_size, tag_to_ix, embedding_dim,
+                 hidden_dim, char_lstm_dim=25, char_to_ix=None,
+                 pre_word_embeds=None, char_embedding_dim=25,):
+
+        super(BiLSTM_CRF, self).__init__()
+        self.embedding_dim = embedding_dim
+        self.hidden_dim = hidden_dim
+        self.vocab_size = vocab_size
+        self.tag_to_ix = tag_to_ix
+        self.tagset_size = len(tag_to_ix)
+        self.out_channels = char_lstm_dim
+
+        print("out_channels: %d, hidden_dim: %d, " % ( char_lstm_dim, hidden_dim))

```

Comment:
Using f strings is prefered now
```suggestion
        print(f"out_channels: {char_lstm_dim}, hidden_dim: {hidden_dim}, ")

```